plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObject.class.php

```diff
@@ -95,6 +111,59 @@ public function recursivelyAddInformationObjects($parentId, $totalRows, $options
     }
   }
 
+  private function updateAllFields($item, $options = array())
+  {
+    $node = new arElasticSearchInformationObjectPdo($item->id, $options);
+    $data = $node->serialize();
+
+    QubitSearch::getInstance()->addDocument($data, 'QubitInformationObject');
+
+    self::$counter++;
+    $this->logEntry($data['i18n'][$data['sourceCulture']]['title'], self::$counter);
+
+    return $node;
+  }
+
+  /**
+   * Update ES fields descendants inherit from their ancestors. This assumes QubitInformationObject atm.
+   */
+  private function updateInheritedFields($item, $options = array())
+  {
+    $node = new arElasticSearchInformationObjectPdo($item->id, $options);
+    $data = array();
+
+    if (isset($options['repository']) && null !== $repository = $node->getRepository())
+    {
+      $data['repository'] = arElasticSearchRepository::serialize($repository);
+    }
+
+    if (isset($options['inheritedCreators']))
+    {
+      foreach ($options['inheritedCreators'] as $creator)
+      {
+        if (!isset($data['inheritedCreators']))
+        {
+          $data['inheritedCreators'] = array();
+        }
+
+        $creatorNode = new arElasticSearchActorPdo($creator->id);
+        $data['inheritedCreators'][] = $creatorNode->serialize();
+      }
+    }
+
+    // Add "Part of" title if this isn't a top level description
+    if (null !== $partOf = $node->getPartOf())
+    {
+      $data['partOf'] = $partOf;
+    }
+
+    // If any of the inherited fields have been set to other than null, do partial update
+    if (array_filter($data))
+    {
+      QubitSearch::getInstance()->partialUpdate(QubitInformationObject::getById($item->id), $data);
```

Comment:
This looks a little overkill in this case, after all the work to improve the population using raw queries, I'd try to avoid having to hydrate the entire IO to do the partial update call. The partial update function only requires the entire resource in the case the document is not found in the index:

https://github.com/artefactual/atom/blob/qa/2.5.x/plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php#L484

I'd change the partialUpdate function, to only require the class of the resource, the id and the partial data. And, in the update fails because the document is not found in the index, I'd do the hydration in there, using the class name and the id before calling the full update.

Reply:
Feel free to disregard this too if running out of time. We can create a new feature ticket to not forgot about the enhancement in the future.